docs/docs/timelock-encryption.md

```diff
@@ -0,0 +1,114 @@
+
+## Use cases
+
+We have identified some cool use cases for timelock encryption, but these are certainly not exhaustive - hopefully you can come up with new, innovative use cases we haven't thought of!
+
+- [responsible vulnerability disclosure](https://timevault.drand.love/)
+- transfer of assets/password upon death
+- sealed bid auctions
+- [MEV prevention](https://coinmarketcap.com/alexandria/glossary/miner-extractable-value-mev)
+- public turn-based games
+- voting
+- quizzes
+
+If you can think of any more cool use cases, join the action on our slack channel and post them in the [#drand-use-cases channel](https://drandworkspace.slack.com/archives/C04UANC4P7D) for discussion!
+
+## The math bit 🤓
+
+### Encryption
+
+A client that wishes to encrypt a message $M \in {0,1}^l$ only decryptable at the epoch $p$ will perform the following:
+
+1. Compute $\mathbb{G_{id}}= e(Q_{id},P) = e(H_1(p),P)$, the "round public key" 
+    - (this can be pre-computed per epoch, it's the same for everyone)
+2. Choose a random $(\theta \in {0,1}^l)$, "the mask"
+3. Set $r = H_3(\theta, M)$ where $H_3:{0,1}^* \to F_q$ is a secure hash function, "the ephemeral secret key"
+4. Output the ciphertext $C = {U, V, W}$ where:
+
+    $U = rG_2\in \mathbb{G_2}$, "the ephemeral public key"
+
+    $V = \theta \oplus H_2(rG_{id})$, "the mask commitment"
+
+    $W = M \oplus H_4(\theta)$, "the one-time pad"
+
+### Decryption
+
+A client that wishes to decrypt a ciphertext $C$ takes the associated signature of epoch $p: \pi_p \in \mathbb{G_1}$ and performs the following:
+
+1. Compute $\theta = V \oplus H_2(e(\pi_p, U))$
+2. Compute $M = W \oplus H_4(\theta)$
+3. Set $r = H_3(\theta, M)$
+4. Test that $U = rG_1 $ if not, reject
+5. $M$ is the corresponding plaintext
+
+
+## Libraries available for timelock encryption
+
+Listed below are various libraries and clients for using timelock encryption. Details on how to install and use them are available at the links provided.
+
+- [tlock](https://github.com/drand/tlock)
+
+tlock is a go CLI and library developed by [Ardan Labs](https://www.ardanlabs.com/) and maintained by the drand team for performing timelock encryption and decryption
+
+- [tlock-js](https://github.com/drand/tlock-js)
+
+tlock-js is a javascript library developed and maintained by the drand team for performing timelock encryption and decryption written in Typescript
+
+- [dee](https://github.com/thibmeu/drand-rs)
+
+dee is a rust CLI developed and maintained by Thibault Meunier from cloudflare. It can perform timelock encryption and decryption, as well as retrieve beacons from any of the drand networks.
```

Comment:
```suggestion
dee is a rust CLI developed and maintained by Thibault Meunier from Cloudflare. It can perform timelock encryption and decryption, as well as retrieve beacons from any of the drand networks.
```